### PR TITLE
Implement Swagger

### PR DESCRIPTION
**Pull request by blancaagc**
Implementation of swagger in the project

### CODE REVIEW

File: src/main/java/com/agilethought/atsceapi/config/ConfigSwagger.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        return new Docket(DocumentationType.SWAGGER_2)
+                .select()
+                .paths(PathSelectors.any())
+                .apis(RequestHandlerSelectors.basePackage("com.agilethought.atsceapi"))
+                .build()
+                .apiInfo(getApiInfo())
+                ;
+
+    }
+
```
Comment (luisgalindoluna) [inline src/main/java/com/agilethought/atsceapi/config/ConfigSwagger.java]: Remember to use Ctrl+Shift+F to format your code.

**Review by luisgalindoluna — CHANGES_REQUESTED** [review]
Overall looks fine, but your branch has conflicts, please resolve those first.

**Comment by ivanflores15** [comment]
image
you have problems with build execution, please fix it.

**Comment by blancaagc** [comment]
PR20 Check proof

Made new commit all checks passed

**Comment by blancaagc** [comment]
PR20 Check proof

Made new commit all checks passed

**Comment by luisgalindoluna** [comment]
@blancaagc Conflicts are still there. In fact you now have an additional conflicting file. 
image

File: src/main/java/com/agilethought/atsceapi/controller/UserController.java [hunk 2]
```diff
@@ -14,37 +18,75 @@
...
-	@PostMapping(value = "/users", consumes = "application/json", produces = "application/json")
-	public NewUserResponse postUser(@RequestBody NewUserRequest request) {
-		return userService.createUser(request);
-	}
+    @ResponseStatus(HttpStatus.OK)
+    @PostMapping(value = "/login")
+    @ApiOperation(value = "Let the user log into the application")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Search in the database without parameters"),
+            @ApiResponse(code = 201, message = "Resource created succefully"),
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/controller/UserController.java]: As far as I know, 201 and 404 is imposible, remove it.

File: src/main/java/com/agilethought/atsceapi/controller/UserController.java [hunk 3]
```diff
@@ -14,37 +18,75 @@
...
-	@GetMapping("/users/{id}")
-	@ResponseStatus(HttpStatus.OK)
-	public UserDTO getUserById(@PathVariable String id){
-		return userService.getUserById(id);
-	}
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping(value = "/users", consumes = "application/json", produces = "application/json")
+    @ApiOperation(value = "Create New User in the application")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Search in the database without parameters"),
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/controller/UserController.java]: 200 is not necessary

File: src/main/java/com/agilethought/atsceapi/controller/UserController.java [hunk 4]
```diff
@@ -14,37 +18,75 @@
...
+    public NewUserResponse postUser(@RequestBody NewUserRequest request) {
+        return userService.createUser(request);
+    }
+
+    @GetMapping("/users/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "Validation the session of the user")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Search in the database without parameters"),
+            @ApiResponse(code = 201, message = "Resource created succefully"),
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/controller/UserController.java]: 201 unnecessary

**Review by ivanflores15 — CHANGES_REQUESTED** [review]

**Comment by luisgalindoluna** [comment]
There's a check failing in this PR as well

**Review by ivanflores15 — APPROVED** [review]
looks good. Great job

**Review by EmiliaGuillen-dev — APPROVED** [review]
looks good

**Review by luisgalindoluna — APPROVED** [review]
Looks good

**Pull request merged**